### DISCUSSION TITLE
CameraView - Access to the live camera feed

### DISCUSSION THREAD

**Discussion by nk-alex** [CommunityToolkit/Maui#1953] [General]
First of all, congratulations for the release of [CameraView](https://github.com/CommunityToolkit/Maui/releases/tag/1.0.0-camera) for MAUI.

This is requirement comes from [Xamarin.Forms](https://github.com/xamarin/XamarinCommunityToolkit/issues/1096). I need to access camera's feed in order to perform object detection in real time. Is this possible in current version? Are you considering it for future releases?

**Comment by bijington** [comment]
Thank you.

I can confirm that what you want to achieve is not possible in v1 however it is something I want to look into. There is a slightly related discussion here https://github.com/CommunityToolkit/Maui/discussions/1687 it is light on details but I hope to try and start defining something for us all to use.

The current key details I have are to provide potentially 2 methods to interact; at the platform specific layer (to avoid unnecessary allocations) and at the shared level (when performance isn't key).

Out of interest which would you be expecting to use for your scenario?

**Reply by nk-alex** [reply]
Hi @bijington thanks for the response!

In my scenario, the object detection task relies on platform specific custom compilations of OpenCV which are accessed from main project via dependency injection. So I could work with both methods. I would just need a way to access live camera feed and a second one to update this camera preview somehow (just to draw a square where the object has been detected)

**Reply by bijington** [reply]
Ok great. I'm thinking something like the following might work

## platform specific

```csharp
public interface IPlatformFrameReceiver
{
#if ANDROID
void OnFrame(Android specific bits);
#endif 
}

public interface IPlatformFrameProvider
{
event EventHandler<FrameEventArgs> OnFrame;
}
```

I'm not overly keen on the event for now but as a v1 for discussion I think it works.

The expectation would be that a developer could register either of the two above and the latter allowing you to customise the frames coming back. 

Then we could have an `IFrameReceiver` on the shared code side. I don't think we could have a provider in the shared code so the approach for rendering things might come down to drawing things on the screen through whatever MAUI offers

**Reply by MartinRobins** [reply]
I too have a scenario that this would suit, as well as potentially not requiring a platform specific handler.
I want to do "edge detection" for documents so that they can be photographed and then automatically de-skewed. In my mind, this can be done in the core application(s) once the image has been captured, but I would want a facility to intercept the preview images and then use these images to then provide an overlay to be displayed in the image preview showing the edges that I have detected.
I have already upvoted your #1687 as I do believe that this would be the most flexible approach, but would like to suggest that it allow feedback as well as interception.

**Reply by bijington** [reply]
I actually forgot I had written this! I'll try and merge this into the other discussion so we have a single location for the details
